pkg/autostart/autostart.go

```diff
@@ -0,0 +1,125 @@
+// Package autostart manage start at login unit files for darwin/linux
+package autostart
+
+import (
+	_ "embed"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/lima-vm/lima/pkg/textutil"
+)
+
+//go:embed lima-vm@INSTANCE.service
+var systemdTemplate string
+
+//go:embed io.lima-vm.autostart.INSTANCE.plist
+var launchdTemplate string
+
+// CreateStartAtLoginEntry respect host OS arch and create unit file
+func CreateStartAtLoginEntry(hostOS, instName, workDir string) error {
+	unitPath := GetStartAtLoginEntryPath(hostOS, instName)
+	if _, err := os.Stat(unitPath); !errors.Is(err, os.ErrNotExist) {
+		return err
+	}
+	tmpl, err := renderTemplate(hostOS, workDir, instName, os.Executable)
```

Comment:
The arguments are in the wrong order:

```suggestion
	tmpl, err := renderTemplate(hostOS, instName, workDir, os.Executable)
```

This results in a broken plist file:

```xml
<dict>
        <key>Label</key>
        <string>io.lima-vm.autostart./Users/jan/.lima/alpine</string>
        …
        <key>WorkingDirectory</key>
        <string>alpine</string>
        …
</dict>
```